dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorArguments.java

```diff
@@ -641,8 +641,8 @@ public List<String> getConfiguration() {
     return getOptions().valuesOf(optionConfiguration);
   }
 
-  public Optional<File> getOutputFile() {
-    return optionAsOptional(optionOutput).filter(file -> !file.getName().isEmpty());
+  public Optional<String> getOutputFile() {
```

Comment:
It is very easy to generate incorrect File instances. So, IMHO the domain class is not providing as much value as you claim here.

Reply:
Yes, I agree that `File` accepts incorrect paths. However, it's even easier to generate incorrect Strings. For example, `new File("a///b").toString()` returns `"a/b"`, meaning that even though I agree with you that `File` is not the best candidate for the domain class, it still does some processing of the input. If you use String, then `a///b` would be allowed, even though it's very likely not what the user wanted.

https://github.com/openjdk-mirror/jdk7u-jdk/blob/f4d80957e89a19a29bb9f9807d2a28351ed7f7df/src/share/classes/java/io/File.java#L250-L253

You could introduce a better domain class, but it may be too much effort for now. I think that `File` or `Path` is enough for the purpose of the code in this class (at least for the local file path case, not the GCS path case).